homeassistant/components/vesync/light.py

```diff
@@ -68,36 +68,57 @@ def __init__(self, device):
     @property
     def brightness(self):
         """Get light brightness."""
-        return round((int(self.device.brightness) / 100) * 255)
+        # get value from pyvesync library api,
+        r = self.device.brightness
+        try:
+            # check for validity of brightness value received
+            brightness_value = int(r)
+            # convert percent brightness to ha expected range
+            brightness_value = round((max(1, brightness_value) / 100) * 255)
+            return brightness_value
+        except ValueError:
+            # deal if any unexpected value
+            _LOGGER.debug(
+                "VeSync - received brightness level from pyvesync api out of range: %d",
+                brightness_value,
+            )
+            return 0
 
     def turn_on(self, **kwargs):
         """Turn the device on."""
-        if ATTR_BRIGHTNESS in kwargs:
-            # get brightness from HA data
-            brightness = int(kwargs[ATTR_BRIGHTNESS])
-            # convert to percent that vesync api expects
-            brightness = round((brightness / 255) * 100)
-            # clamp to 1-100
-            brightness = max(1, min(brightness, 100))
-            self.device.set_brightness(brightness)
+        # set brightness level
+        if self.color_mode in (COLOR_MODE_BRIGHTNESS, COLOR_MODE_COLOR_TEMP):
+            if ATTR_BRIGHTNESS in kwargs:
+                # get brightness from HA data
+                brightness = int(kwargs[ATTR_BRIGHTNESS])
+                # convert to percent that vesync api expects
+                brightness = round((max(1, brightness) / 255) * 100)
+                # ensure value between 0-100
+                brightness = max(1, min(brightness, 100))
+                self.device.set_brightness(brightness)
+                return
```

Comment:
Both brightness and color temp can come in the same turn on all. Can the device set both?

Reply:
I changed the logic so you can set both attributes simultaneuosly. Good catch for noticing that!